src/auth/auth.service.ts

```diff
@@ -0,0 +1,102 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+import { PrismaService } from 'src/database/prisma.service';
+import { CreateUserDto } from './input/auth.dto';
+import { Tokens } from './models/tokens.type';
+import * as bcrypt from 'bcrypt';
+import { JwtPayload } from './models/jwt-payload.type';
+import { UsersService } from '../users/users.service';
+import { DatabaseRecord } from '../common/models/databaseRecord.model';
+import { User } from '../users/models/user.type';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService,
+    private config: ConfigService,
+    private usersService: UsersService,
+  ) {}
+
+  async register(dto: CreateUserDto): Promise<Tokens> {
+    const user: DatabaseRecord<User> = await this.usersService.createUser(dto);
+
+    const tokens = await this.getTokens(user.id, user.email);
+    await this.updateRtHash(user.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  async login(dto: CreateUserDto): Promise<Tokens> {
+    const user = await this.usersService.getUserByEmail(dto.email);
+
+    if (!user) throw new ForbiddenException('Access Denied');
+
+    const passwordMatches = await bcrypt.compare(dto.password, user.hash);
+    if (!passwordMatches) throw new ForbiddenException('Access Denied');
+
+    const tokens = await this.getTokens(user.id, user.email);
+    await this.updateRtHash(user.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  async logout(userId: string): Promise<boolean> {
+    await this.prisma.user.updateMany({
+      where: {
+        id: userId,
+        hashedRt: {
+          not: null,
+        },
+      },
+      data: {
+        hashedRt: null,
+      },
+    });
+    return true;
+  }
```

Comment:
Aside from `hashedRt` that we discussed yesterday - do we have to `return true` here? How about just awaiting the method and returning `Promise<void>` or just returning the method? If it returns then it was successful - if it throws error then it failed.